ps1/ps1/test/twitter/ExtractTest.java

```diff
@@ -1,178 +0,0 @@
-/* Copyright (c) 2007-2016 MIT 6.005 course staff, all rights reserved.
- * Redistribution of original or derived work requires permission of course staff.
- */
-package twitter;
-
-import static org.junit.Assert.*;
-
-import java.time.Instant;
-import java.util.Arrays;
-import java.util.Set;
-
-import org.junit.Test;
-
-public class ExtractTest {
-
-    /*
-     * TODO: your testing strategies for these methods should go here. See the
-     * ic03-testing exercise for examples of what a testing strategy comment
-     * looks like. Make sure you have partitions.
-     * 
-     * Partitions for timespan:
-     * later date first
-     * later date after
-     * single tweet
-     * two tweets
-     * multiple tweets
-     * contains two tweets with the same date
-     * 
-     * Partitions for getusername:
-     * no tweets have username
-     * all tweets have username
-     * some tweets have username
-     * 
-     * within this space:
-     * username at start
-     * username at end
-     * username in middle
-     * multiple usernames
-     * one username
-     * no username
-     * two usernames in a row
-     * same usernames in one tweet
-     * same usernames in multiple tweets
-     * 
-     */
-
-    private static final Instant d1 = Instant.parse("2016-02-17T10:00:00Z");
-    private static final Instant d2 = Instant.parse("2016-02-17T11:00:00Z");
-    private static final Instant d3 = Instant.parse("2016-02-17T11:00:00Z");
-
-    private static final Tweet tweet1 = new Tweet(1, "alyssa", "is it reasonable to talk about rivest so much?", d1);
-    private static final Tweet tweet2 = new Tweet(2, "bbitdiddle", "rivest talk in 30 minutes #hype", d2);
-    private static final Tweet tweet3 = new Tweet(3, "goobergoo", "@firstmention is a real dunce", d3);
-    private static final Tweet tweet4 = new Tweet(4, "ginallygoo", "you know a dunce @ parties? @lastmention", d3);
-    private static final Tweet tweet5 = new Tweet(5, "goofergoob", "@lastmention, @lastmention, where art thou", d3);
-    private static final Tweet tweet6 = new Tweet(6, "goo-boo", "@donald_j @trump @good-bye", d3);
-    private static final Tweet tweet7 = new Tweet(7, "goober_grub", "@donnie!! @trump @getout#now", d3);
```

Comment:
Oh smart... I didn't think about something with a space. But the way I wrote my method, it just keeps adding characters to a given username until an illegal character is found (and @ is an illegal character). So I think it would pass but I should check that